Review.swift

```diff
@@ -0,0 +1,294 @@
+                                                return
+                                            }
+                                            
+                                            logger.info("Setting Job Status...")
+                                            job.status = .done
+                                            job.update(on: req.db).map {
+                                                logger.info("Reviewer Job ended successful!")
+                                            }
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
+
+extension Review {
+    func deleteDirectory(pullRequest: PullRequest) {
+        let dirs = DirectoryConfiguration.self
+        let path = dirs.detect().workingDirectory
+        let fileManager = FileManager()
+        
+        let shortHash = pullRequest.latestCommit.prefix(8)
+        
+        do {
+            logger.info("Deleting directory: sources_\(shortHash)")
+            try fileManager.removeItem(atPath: path + "sources_" + shortHash)
+        } catch {
+            logger.info("\(error.localizedDescription)")
+        }
+    }
+    
+    func directoryExists(pullRequest: PullRequest) -> Bool {
+        let dirs = DirectoryConfiguration.self
+        let path = dirs.detect().workingDirectory
+        let fileManager = FileManager()
+        
+        let shortHash = pullRequest.latestCommit.prefix(8)
+        
+        let directoryExists = fileManager.fileExists(atPath: path + "sources_" + shortHash)
+        logger.info("Directory exists: \(directoryExists.description)")
+        
+        return directoryExists
+    }
+    
+    func createDirectory(pullRequest: PullRequest) {
+        let dirs = DirectoryConfiguration.self
+        let path = dirs.detect().workingDirectory
+        let fileManager = FileManager()
+        
+        let shortHash = pullRequest.latestCommit.prefix(8)
+        
+        do {
+            logger.info("Creating directory: sources_\(shortHash)")
+            try fileManager.createDirectory(atPath: path + "sources_" + shortHash, withIntermediateDirectories: false, attributes: nil)
+        } catch {
+            logger.info("\(error.localizedDescription)")
+        }
+    }
+    
+    func unzipSources(fileName: String, fileExtension: String, pullRequest: PullRequest) {
+        let dirs = DirectoryConfiguration.self
+        let path = dirs.detect().workingDirectory
+        let fileManager = FileManager()
+        
+        let shortHash = pullRequest.latestCommit.prefix(8)
+        
+        do {
+            logger.info("Unzip sources...")
+            try fileManager.unzipItem(at: URL(fileURLWithPath: path + "sources_" + shortHash + "/" + fileName + fileExtension), to: URL(fileURLWithPath: path + "sources_" + shortHash))
```

Comment:
🚫 Error
 [Line Length] At line 222: Line should be 150 characters or less: currently 184 characters